core/token.go

```diff
@@ -225,6 +227,44 @@ func (ts TokenService) ValidateIDToken(ctx *HttpContext, idToken string) {
 	}
 }
 
+/*
+   Extract the RoleSessionName as the username which is in the 'sub' claim of the token.
+   Function should return the user ID (string containing the <username>@<tenant>) or an
+   empty string if an error of any kind.
+*/
+func (ts TokenService) ExtractUserIDFromIDToken(ctx *HttpContext, idToken string) (string) {
+	if idToken == "" {
+		ctx.Log.Err("No ID token provided.")
+		return ""
+	}
+
+	//Fetch the public key
+	publicKey,err := ts.GetPublicKeyPEM(ctx)
+	if err != nil {
+		ctx.Log.Err(fmt.Sprintf("Could not fetch public key: %v\n", err))
+		return ""
+	}
+
+	//Parse takes the token string and a function for looking up the public key
+	token,err:=jwt.Parse(idToken,func(token*jwt.Token)(interface{}, error) {
```

Comment:
`go fmt` wasn't probably run here. seems like there should be spaces around `:=`